logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/message/DeleteForMeHandler.kt

```diff
@@ -0,0 +1,35 @@
+package com.wire.kalium.logic.sync.receiver.message
+
+import com.wire.kalium.logger.KaliumLogger
+import com.wire.kalium.logic.data.conversation.ConversationRepository
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.message.Message
+import com.wire.kalium.logic.data.message.MessageContent
+import com.wire.kalium.logic.data.message.MessageRepository
+import com.wire.kalium.logic.data.user.UserRepository
+import com.wire.kalium.logic.kaliumLogger
+
+class DeleteForMeHandler(
+    private val conversationRepository: ConversationRepository,
+    private val messageRepository: MessageRepository,
+    private val userRepository: UserRepository
+) {
+
+    suspend fun handle(
+        message: Message,
+        messageContent: MessageContent.DeleteForMe
+    ){
+        // The conversationId comes with the hidden message[content] only carries the conversationId VALUE,
+        // we need to get the DOMAIN from the self conversationId[here is the message.conversationId]
+        val conversationId = messageContent.conversationId ?:
+        ConversationId(messageContent.unqualifiedConversationId, userRepository.getSelfUserId().domain)
```

Comment:
Why not do this inside the `ProtobufContentMapper` as I mentioned in another comment.